frontend/src/pages/EventCreationPage/components/EventCreationForm/index.tsx

```diff
@@ -29,9 +29,29 @@ interface InitialValuesType {
   hosts: OfficerNameData[];
 }
 
+function findNearestHour(date, isStartDate) {
+  let hour = date.getHours();
+  let startMinutes = date.getMinutes();
+  if (startMinutes === 0) {
+    startMinutes = '00';
+  } else if (startMinutes <= 30) {
+    startMinutes = '30';
+  } else if (startMinutes > 30) {
+    startMinutes = '00';
+    hour += 1;
```

Comment:
It is much better to use **date-fns** (the datetime library we are using) to handle datetime arithmetic like what you're doing here, since it does all the overflow handling for you. Reference: https://date-fns.org/docs/Getting-Started

Specific function to add some amount of time to a specific date: https://date-fns.org/v2.21.1/docs/add